### DISCUSSION TITLE
header encoding issue???

### DISCUSSION THREAD

**Discussion by BrightArgo** [encode/httpx#3044] [Potential Issue]
receiving the following error:

venv/lib64/python3.11/site-packages/httpx/_utils.py", line 53, in normalize_header_value
    return value.encode(encoding or "ascii")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'ascii' codec can't encode character '\u2018' in position 7: ordinal not in range(128)

I am not directly calling the httpx module, it is being referenced by gradio which in turns passes on the request headers via the browser.  I believe that is where the breakdown occurs.  My understanding of the code and from forums is that by default UTF-8 should be the encoding utilized.  If this is the case, I am unclear why httpx falls back to using ascii.  I am running Python 3.11 on Oracle Linux 8 and have doublechecked all my environment variables.  My code works fine on my Mac.  Happy to provide additional details.

**Comment by BrightArgo** [comment]
Replaced ascii in _utils.py with UTF-8 and discovered the real issue related to header encoding.

**Reply by bangursandeep** [reply]
what was the real issue ? If you could enlighten please .

**Reply by BrightArgo** [reply]
string for API key needed to be quoted.
